src/Protocol.cpp

```diff
@@ -0,0 +1,69 @@
+#include "pipert/Protocol.h"
+#include "pipert/UDPConnection.h"
+
+#include <cassert>
+#include <string.h>
+
+namespace pipert {
+
```

Comment:
Every template in this file should go into the header file instead, because it will be not possible to compile the code at places where the header is included only. Currently, somehow the whole code compiles, but an ODR error is detected at compilation. This means that the only reason it is compiling is that somehow, the cpp is compiled multiple times. As this error was caught only by address sanitizer, it is possible that it causes tests failing on the CI where we skip asan.

Reply:
Nice catch, thank you. I have already experienced before in local environment that creating a network related channel (which includes the Protocol.h file) leads to compiler errors, but I thought it's occurred by an improper CMake config. Now CI checks pass again.